homeassistant/components/asuswrt/config_flow.py

```diff
@@ -76,25 +74,30 @@ def _show_setup_form(self, user_input=None, errors=None):
         if user_input is None:
             user_input = {}
 
+        schema = {
+            vol.Required(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
+            vol.Required(CONF_USERNAME, default=user_input.get(CONF_USERNAME, "")): str,
+            vol.Optional(CONF_PASSWORD)
+            if self.show_advanced_options
+            else vol.Required(CONF_PASSWORD): str,
+            vol.Required(CONF_PROTOCOL, default=PROTOCOL_SSH): vol.In(
+                {PROTOCOL_SSH: "SSH", PROTOCOL_TELNET: "Telnet"}
+            ),
+        }
+
+        if self.show_advanced_options:
+            schema[vol.Required(CONF_PORT, default=0)] = vol.All(
```

Comment:
Wouldn't marking it `vol.Optional` and no default make more sense to indicate we should figure it out ourselves? (not sure if we handle default numbers well though)

Reply:
I was convinced that `vol.Optional` with numbers didn't work, but that's not true. Just tested now and work correctly, so I'm going to modify this as suggested by you.